[Bug]: Investigate failing cuDF timezone tests

### Describe the bug

There are currently a few tests failing for timezones and cuDF:

```

tests/utils.py:60: AssertionError
=========================== short test summary info ============================
FAILED tests/expr_and_series/convert_time_zone_test.py::test_convert_time_zone_from_none[cudf_constructor] - AssertionError: ('2020-01-01T05:45+0000', '2020-01-01T05:45+0545')
FAILED tests/expr_and_series/convert_time_zone_test.py::test_convert_time_zone[cudf_constructor] - AssertionError: ('2020-01-01T05:45+0000', '2020-01-01T05:45+0545')
FAILED tests/expr_and_series/convert_time_zone_test.py::test_convert_time_zone_series[cudf_constructor] - AssertionError: ('2020-01-01T05:45+0000', '2020-01-01T05:45+0545')
FAILED tests/expr_and_series/replace_time_zone_test.py::test_replace_time_zone_series[cudf_constructor] - AssertionError: ('2020-01-01T00:00+0000', '2020-01-01T00:00+0545')
FAILED tests/expr_and_series/replace_time_zone_test.py::test_replace_time_zone[cudf_constructor] - AssertionError: ('2020-01-01T00:00+0000', '2020-01-01T00:00+0545')
===== 5 failed, 4842 passed, 15 skipped, 143 xfailed in 108.35s (0:01:48) ======
```

### Steps or code to reproduce the bug

Run the cuDF testing notebook;
https://www.kaggle.com/code/marcogorelli/testing-cudf-in-narwhals

### Expected results

No error

### Actual results

Errors on five tests

### Please run narwhals.show_version() and enter the output below.

```shell
Version on main
```


### Relevant log output

_No response_

--- a/tests/expr_and_series/convert_time_zone_test.py
+++ b/tests/expr_and_series/convert_time_zone_test.py
@@ -17,8 +17,13 @@ from tests.utils import is_windows
 def test_convert_time_zone(
     constructor: Constructor, request: pytest.FixtureRequest
 ) -> None:
-    if (any(x in str(constructor) for x in ("pyarrow", "modin")) and is_windows()) or (
-        "pandas_pyarrow" in str(constructor) and parse_version(pd.__version__) < (2, 1)
+    if (
+        (any(x in str(constructor) for x in ("pyarrow", "modin")) and is_windows())
+        or (
+            "pandas_pyarrow" in str(constructor)
+            and parse_version(pd.__version__) < (2, 1)
+        )
+        or ("cudf" in str(constructor))
     ):
         request.applymarker(pytest.mark.xfail)
     data = {
@@ -41,10 +46,12 @@ def test_convert_time_zone_series(
     constructor_eager: Any, request: pytest.FixtureRequest
 ) -> None:
     if (
-        any(x in str(constructor_eager) for x in ("pyarrow", "modin")) and is_windows()
-    ) or (
-        "pandas_pyarrow" in str(constructor_eager)
-        and parse_version(pd.__version__) < (2, 1)
+        (any(x in str(constructor_eager) for x in ("pyarrow", "modin")) and is_windows())
+        or (
+            "pandas_pyarrow" in str(constructor_eager)
+            and parse_version(pd.__version__) < (2, 1)
+        )
+        or ("cudf" in str(constructor_eager))
     ):
         request.applymarker(pytest.mark.xfail)
     data = {
@@ -73,6 +80,7 @@ def test_convert_time_zone_from_none(
             and parse_version(pd.__version__) < (2, 1)
         )
         or ("pyarrow_table" in str(constructor) and parse_version(pa.__version__) < (12,))
+        or ("cuDF" in str(constructor))
     ):
         request.applymarker(pytest.mark.xfail)
     if "polars" in str(constructor) and parse_version(pl.__version__) < (0, 20, 7):

--- a/tests/expr_and_series/convert_time_zone_test.py
+++ b/tests/expr_and_series/convert_time_zone_test.py
@@ -80,7 +80,7 @@ def test_convert_time_zone_from_none(
             and parse_version(pd.__version__) < (2, 1)
         )
         or ("pyarrow_table" in str(constructor) and parse_version(pa.__version__) < (12,))
-        or ("cuDF" in str(constructor))
+        or ("cudf" in str(constructor))
     ):
         request.applymarker(pytest.mark.xfail)
     if "polars" in str(constructor) and parse_version(pl.__version__) < (0, 20, 7):

--- a/tests/expr_and_series/replace_time_zone_test.py
+++ b/tests/expr_and_series/replace_time_zone_test.py
@@ -20,6 +20,7 @@ def test_replace_time_zone(
         (any(x in str(constructor) for x in ("pyarrow", "modin")) and is_windows())
         or ("pandas_pyarrow" in str(constructor) and parse_version(pd.__version__) < (2,))
         or ("pyarrow_table" in str(constructor) and parse_version(pa.__version__) < (12,))
+        or ("cudf" in str(constructor))
     ):
         request.applymarker(pytest.mark.xfail)
     data = {
@@ -76,6 +77,7 @@ def test_replace_time_zone_series(
             "pyarrow_table" in str(constructor_eager)
             and parse_version(pa.__version__) < (12,)
         )
+        or ("cudf" in str(constructor_eager))
     ):
         request.applymarker(pytest.mark.xfail)
     data = {

--- a/tests/series_only/to_dummy_test.py
+++ b/tests/series_only/to_dummy_test.py
@@ -18,11 +18,7 @@ def test_to_dummies(constructor_eager: Any, sep: str) -> None:
 
 
 @pytest.mark.parametrize("sep", ["_", "-"])
-def test_to_dummies_drop_first(
-    request: pytest.FixtureRequest, constructor_eager: Any, sep: str
-) -> None:
-    if "cudf" in str(constructor_eager):
-        request.applymarker(pytest.mark.xfail)
+def test_to_dummies_drop_first(constructor_eager: Any, sep: str) -> None:
     s = nw.from_native(constructor_eager({"a": data}), eager_only=True)["a"].alias("a")
     result = s.to_dummies(drop_first=True, separator=sep)
     expected = {f"a{sep}2": [0, 1, 0], f"a{sep}3": [0, 0, 1]}

